rpcs3/Emu/RSX/rsx_methods.cpp

```diff
@@ -1444,6 +1471,15 @@ namespace rsx
 			const auto write_address = get_address(dst_offset, dst_dma, HERE);
 			const auto data_length = in_pitch * (line_count - 1) + line_length;
 
+			if (const auto [shader_offset, shader_dma] = method_registers.shader_program_address();
```

Comment:
This code is repeated all over the code. Refactor it into one RSX method to check and invalidate.
`RSXThread::invalidate_fragment_program(const address_range&)`